gelbooru-overhaul.user.js

```diff
@@ -9,485 +9,1551 @@
+                    li.appendChild(pText);
+                    break;
+                default:
+                    throw new TypeError(`Unknown type ${type} of ${pref}`);
+            }
+            return li;
+        }
+        function updateInputValues() {
+            document.querySelectorAll("#goConfigWindow input:not([type='submit'])").forEach(/** @param {HTMLInputElement} i */ i => {
+                switch (i.type) {
+                    case "checkbox":
+                        i.checked = resolve(i.id + ".value", configManager.config);
+                        break;
+                
+                    case "text":
+                        i.value = resolve(i.id + ".value", configManager.config);
+                        break;
                 }
+            });
+        }
+    }
+    /**
+     * Current page type (see {@link PageTypes})
+     * @returns {string}
+     */
+    function getPageType() {
+        let params = new URLSearchParams(document.URL.split('?')[1]);
+
+        if (!params.has("page"))
+            return PageTypes.UNDEFINED;
+
+        if (params.get("page") == "post" && params.get("s") == "list")
+            return PageTypes.GALLERY;
+
+        if (params.get("page") == "post" && params.get("s") == "view")
+            return PageTypes.POST;
+
+        if (params.get("page") == "wiki" && params.get("s") == "view")
+            return PageTypes.WIKI_VIEW;
+
+        if (params.get("page") == "pool" && params.get("s") == "show")
+            return PageTypes.POOL_VIEW;
+
+        return PageTypes.UNDEFINED;
+    }
+    /** 
+     * @typedef PostItem
+     * @property {string} highResThumb - high resolution thumb url (image/animated gif/video preview)
+     * @property {string} download - download url (original image/gif/video)
+     * @property {Object} tags - list of tags by category
+     * @property {string[]} tags.artist - artist tags (can be empty)
+     * @property {string[]} tags.character - character tags (can be empty)
+     * @property {string[]} tags.copyright - copyright tags (can be empty)
+     * @property {string[]} tags.metadata - metadata tags (can be empty)
+     * @property {string[]} tags.general - general tags (can be empty)
+     * @property {string} md5 - md5 for file (0's for video)
+     * @property {number} id - post id
+     */
+    /**
+     * Cache and return post item
+     * @param {number} postId 
+     * @returns {Promise<PostItem>}
+     */
+    function loadPostItem(postId) {
+        return new Promise((resolve, reject) => {
+            /** @type {Object<number, PostItem} */
+            let postCache = GM_getValue("postCache", {});
+
+            // just clear postCache if exceeded limit
+            if (Object.keys(postCache).length > configManager.config.general.items.maxCache.value)
+                postCache = {};
+
+            if (!postCache[postId]) {
+                fetch("https://gelbooru.com/index.php?page=post&s=view&id=" + postId)
```

Comment:
```suggestion
                fetch("https://" + window.location.host + "/index.php?page=post&s=view&id=" + postId)
```
Add support for ja.gelbooru.com